### PR TITLE
BE/feature/form-fields

### PR DESCRIPTION
**Pull request by rburaksaritas**
Form field endpoints can be reached at:
```
GET api/form_fields/resource
GET api/form_fields/need
GET api_form_fields/subtype/{subtype_name}
```
Returns json in format:
```{
    "fields": [
        {
            "name": "size",
            "label": "Size",
            "type": "text"
        },
        {
            "name": "gender",
            "label": "Gender",
            "type": "text"
        },
        {
            "name": "age",
            "label": "Age",
            "type": "text"
        },
        {
            "name": "subtype",
            "label": "Subtype",
            "type": "text"
        }
    ]
}
```

### CODE REVIEW

**Review by mervegrbz — APPROVED** [review]
Thanks a lot, this will help a lot to me

**Review by bgmrsln — COMMENTED** [review]
Can you check my comments

File: Disaster-Response-Platform/backend/Controllers/form_fields.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            {"name": "condition", "label": "Condition", "type": "select", "options": ["new", "used"]},
+            {"name": "initialQuantity", "label": "Initial Quantity", "type": "number"},
+            {"name": "currentQuantity", "label": "Current Quantity", "type": "number"},
+            {"name": "type", "label": "Type", "type": "text"},
+            {"name": "x", "label": "X Coordinate", "type": "number"},
+            {"name": "y", "label": "Y Coordinate", "type": "number"}
+        ]
+    },
+    "need": {
+        "fields": [
```
[the remark this answers is not in the document]
Comment (rburaksaritas) [reply Disaster-Response-Platform/backend/Controllers/form_fields.json:15]: > I remember mentioning a last_updated field to show users when last entry is updated. Do we eliminate it?

I added it in the BE/feature/need-dates & BE/feature/resource-dates branches, you can check them from currently open PRs. However, this json is for form-generation, but last_updated field is updated in the service category automatically when an update call is received. So I did not think it was necessary to include it here. Similarly, created_by fields should also probably be avoided.

File: Disaster-Response-Platform/backend/Controllers/form_fields.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
+{
+    "resource": {
+        "fields": [
+            {"name": "_id", "label": "ID", "type": "text"},
+            {"name": "created_by", "label": "Created By", "type": "text"},
+            {"name": "condition", "label": "Condition", "type": "select", "options": ["new", "used"]},
```
[the remark this answers is not in the document]
Comment (rburaksaritas) [reply Disaster-Response-Platform/backend/Controllers/form_fields.json:6]: > what is condition

it was in the class diagram, probably for cloth like resources like its new unopened or "2. el". :)

**Pull request merged**